### PR TITLE
Grpc struct 1

### PR DESCRIPTION
**Pull request by stasBigunenko**
GRPC update existing server #53

```
service UserService {
  rpc GetUser(UserID) returns (Obj) {}
  rpc CreateUser(Username) returns (Obj) {}
  rpc UpdateUser(Obj) returns (Obj) {}
  rpc DeleteUser(UserID) returns (google.protobuf.Empty) {}
}
```

### CODE REVIEW

File: pkg/grpc/proto/grpcservice.proto [hunk 1]
```diff
@@ -3,21 +3,25 @@ syntax = "proto3";
...
+  rpc UpdateUser(Obj) returns (Obj) {}
+  rpc DeleteUser(UserID) returns (google.protobuf.Empty) {}
 }
 
 message Obj {
   string id = 1;
-  string title = 2;
-  string description = 3;
+  string name = 2;
 }
```
Comment (DrIhor) [inline pkg/grpc/proto/grpcservice.proto:19]: One of params should be ctx value.

**Review by DrIhor — APPROVED** [review]

**Comment by stasBigunenko** [comment]
Made two proto files

user
```
service UserGRPCService {
  rpc Get(Id) returns (Obj) {}
  rpc Create(Name) returns (Obj) {}
  rpc Update(Obj) returns (Obj) {}
  rpc Delete(Id) returns (google.protobuf.Empty) {}
}
```
and account
```
service AccountGRPCService {
  rpc Get(User_id) returns (Accounts) {}
  rpc Create(User_id) returns (Obj) {}
  rpc Update(Balance) returns (Balance) {}
  rpc Delete(User_id) returns (google.protobuf.Empty) {}
}
```

File: pkg/accountGRPC/account.proto [hunk 2]
```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package accountGRPC;
+option go_package = "github.com/stasBigunenko/monorepa/pkg/account/proto";
+
+import "google/protobuf/empty.proto";
+
+service UserGRPCService {
+  rpc Get(User_id) returns (Accounts) {}
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: Rename this to GetUserAccounts

File: pkg/accountGRPC/account.proto [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  rpc Create(User_id) returns (Obj) {}
+  rpc Update(Balance) returns (Balance) {}
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: It would be logically more clear if we update Account, not Balance

File: pkg/accountGRPC/account.proto [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  rpc Delete(User_id) returns (google.protobuf.Empty) {}
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: Refactor Delete(Account). The func you currently have will delete all user accounts and this is not what we want.

File: pkg/accountGRPC/account.proto [hunk 5]
```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package accountGRPC;
+option go_package = "github.com/stasBigunenko/monorepa/pkg/account/proto";
+
+import "google/protobuf/empty.proto";
+
+service UserGRPCService {
+  rpc Get(User_id) returns (Accounts) {}
+  rpc Create(User_id) returns (Obj) {}
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: Add rpc for GetAccount(id)

File: pkg/accountGRPC/account.proto [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  rpc Update(Balance) returns (Balance) {}
+  rpc Delete(User_id) returns (google.protobuf.Empty) {}
+}
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: Add RPC for ListAccounts to display all accounts of all users.

File: pkg/accountGRPC/account.proto [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+message Obj {
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: I'd rather rename this to Account

File: pkg/accountGRPC/account.proto [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  string id = 1;
+  string user_id = 2;
+  string balance = 3;
+}
+
+message User_id {
+  string user_id = 1;
+}
+
+message Balance {
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: Why do you need this?

File: pkg/accountGRPC/account.proto [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  string id = 1;
+  string balance = 2;
+}
+
+message Accounts {
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: repeated Account accounts = 1

File: pkg/grpc/proto/grpcservice.proto [hunk 10]
```diff
@@ -3,21 +3,25 @@ syntax = "proto3";
...
-}
 
-message Items {
-  repeated Obj items = 1;
+service UserService {
+  rpc GetUser(UserID) returns (Obj) {}
+  rpc CreateUser(Username) returns (Obj) {}
+  rpc UpdateUser(Obj) returns (Obj) {}
+  rpc DeleteUser(UserID) returns (google.protobuf.Empty) {}
 }
```
Comment (alex-a-renoire) [inline pkg/grpc/proto/grpcservice.proto:14]: add ListUsers

File: pkg/grpc/proto/grpcservice.proto [hunk 11]
```diff
@@ -3,21 +3,25 @@ syntax = "proto3";
...
 
 message Obj {
   string id = 1;
```
Comment (alex-a-renoire) [inline pkg/grpc/proto/grpcservice.proto:17]: Obj is a meaningless name for a struct. Call it User. GetUser should return User, not Obj.

File: pkg/accountGRPC/account.proto (hunk 3 above)
Comment (alex-a-renoire) [reply pkg/accountGRPC/account.proto]: UpdateAccount(Account) returns (Account)

File: pkg/accountGRPC/account.proto (hunk 4 above)
Comment (alex-a-renoire) [reply pkg/accountGRPC/account.proto]: rename DeleteAccount(AccountID)

File: pkg/accountGRPC/account.proto [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  rpc Delete(User_id) returns (google.protobuf.Empty) {}
+}
+
+message Obj {
+  string id = 1;
+  string user_id = 2;
+  string balance = 3;
+}
+
+message User_id {
```
Comment (alex-a-renoire) [inline pkg/accountGRPC/account.proto]: Use EITHER sankecase or camelcase, never mix. Here better use camelcase, in fact. Rename: UserID

File: pkg/accountGRPC/account.proto (hunk 2 above)
Comment (stasBigunenko) [reply pkg/accountGRPC/account.proto]: done

File: pkg/accountGRPC/account.proto (hunk 3 above)
Comment (stasBigunenko) [reply pkg/accountGRPC/account.proto]: done

File: pkg/accountGRPC/account.proto (hunk 4 above)
Comment (stasBigunenko) [reply pkg/accountGRPC/account.proto]: done

File: pkg/accountGRPC/account.proto (hunk 5 above)
Comment (stasBigunenko) [reply pkg/accountGRPC/account.proto]: done

File: pkg/accountGRPC/account.proto (hunk 6 above)
Comment (stasBigunenko) [reply pkg/accountGRPC/account.proto]: done

File: pkg/accountGRPC/account.proto (hunk 8 above)
Comment (stasBigunenko) [reply pkg/accountGRPC/account.proto]: We need this to update the balance of each account of the user

File: pkg/accountGRPC/account.proto (hunk 9 above)
Comment (stasBigunenko) [reply pkg/accountGRPC/account.proto]: done

File: pkg/grpc/proto/grpcservice.proto (hunk 10 above)
Comment (stasBigunenko) [reply pkg/grpc/proto/grpcservice.proto:14]: done

File: pkg/grpc/proto/grpcservice.proto (hunk 11 above)
Comment (stasBigunenko) [reply pkg/grpc/proto/grpcservice.proto:17]: done

File: pkg/accountGRPC/account.proto (hunk 12 above)
Comment (stasBigunenko) [reply pkg/accountGRPC/account.proto]: fixed

**Review by alex-a-renoire — APPROVED** [review]

**Pull request merged**